Core/SuppressedCompatWarningIdentifiers.cs

```diff
@@ -0,0 +1,34 @@
+using System.IO;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using CKAN.Versioning;
+
+namespace CKAN
+{
+    class SuppressedCompatWarningIdentifiers
+    {
+        public GameVersion GameVersionWhenWritten;
+        public HashSet<string> Identifiers = new HashSet<string>();
+
+        public static SuppressedCompatWarningIdentifiers LoadFrom(GameVersion gameVer, string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                return new SuppressedCompatWarningIdentifiers() { GameVersionWhenWritten = gameVer };
+            }
+            var saved = JsonConvert.DeserializeObject<SuppressedCompatWarningIdentifiers>(File.ReadAllText(filename));
```

Comment:
What about catching `JsonException` here in case the file got corrupted?
It's probably insignificant enough not to bother with showing any warning about this to the user (but maybe write a line to the log?)

Reply:
Nice idea, consolidating into one exception handler lets us remove the `File.Exists` special case check and combine the "empty" return statements. Done.